openslides_backend/action/actions/user/conditional_speaker_cascade_mixin.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Dict
+
+from ....services.datastore.commands import GetManyRequest
+from ....shared.patterns import fqid_from_collection_and_id
+from ...action import Action
+from ..speaker.delete import SpeakerDeleteAction
+
+
+class ConditionalSpeakerCascadeMixin(Action):
+    """
+    Mixin for user actions that deletes unstarted speeches of users that were either deleted, or removed from a meeting
+    """
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        removed_meeting_id = self.get_removed_meeting_id(instance)
+        if removed_meeting_id is not None:
+            delete_all = removed_meeting_id == 0
+            user = self.datastore.get(
```

Comment:
You could read in1 call and only the needed ones with:
```python
filter_ = FilterOperator("user_id", "=", instance["id"])
if removed_meeting_id:
    filter_ = And(filter_, FilterOperator("meeting_id", "=", removed_meeting_id)
meeting_user_ids = self.datastore.filter("meeting_user", filter_, ["speaker_ids"])
```
and you read only the needed meeting_user_ids in 1 call